src/libraries/Microsoft.PowerFx.Core/Functions/FunctionScopeInfo.cs

```diff
@@ -107,11 +107,23 @@ public virtual bool CheckInput(Features features, TexlNode inputNode, DType inpu
                 }
             }
             else if (_function.ParamTypes[0].IsTable)
-            {
-                if (!typeScope.IsTable)
-                {
-                    errors.Error(callNode, TexlStrings.ErrNeedTable_Func, _function.Name);
-                    fArgsValid = false;
+            {
+                if (features.PowerFxV1CompatibilityRules)
+                {
+                    // Untyped blank values should not be used to define the scope
+                    if (!typeScope.IsTableNonObjNull)
+                    {
+                        errors.Error(callNode, TexlStrings.ErrNeedTable_Func, _function.Name);

```

Comment:
Could we combine the condition so we don't duplicate this block?

`if ((features.PowerFxV1CompatibilityRules && !typeScope.IsTableNonObjNull) || 
      !typeScope.IsTable)`?

Reply:
Updated it